packages/bottom-tabs/src/views/BottomTabBar.tsx

```diff
@@ -249,18 +260,87 @@ export default function BottomTabBar({
     descriptors,
     dimensions,
     layout,
+    tabBarPosition,
   });
 
   const tabBarBackgroundElement = tabBarBackground?.();
 
+  const tabBarPositionMap: TabBarPositionMapConfig = {
+    right: {
+      tabBarItem: {
+        height:
+          Platform.OS === ('ios' || 'android')
+            ? tabBarHeight / 1.5
+            : DEFAULT_TABBAR_HEIGHT,
+        marginTop: COMPACT_TABBAR_HEIGHT / 2,
+        flex: Platform.OS === ('ios' || 'android') ? 0 : 0.07,
+      },
+      tabBar: {
+        paddingHorizontal: 5,
+        borderLeftColor: colors.border,
+        paddingTop: insets.top + tabBarHeight,
+        borderLeftWidth: StyleSheet.hairlineWidth,
+      },
+      content: {
+        flex: 1,
+        width: tabBarHeight,
+        maxWidth:
+          Platform.OS === ('ios' || 'android')
+            ? DEFAULT_TABBAR_HEIGHT
+            : DEFAULT_TABBAR_HEIGHT * 1.3,
+      },
+    },
+
+    left: {
+      tabBarItem: {
+        height:
+          Platform.OS === ('ios' || 'android')
+            ? tabBarHeight / 1.5
+            : DEFAULT_TABBAR_HEIGHT,
+        marginTop: COMPACT_TABBAR_HEIGHT / 2,
+        flex: Platform.OS === ('ios' || 'android') ? 0 : 0.07,
+      },
+      tabBar: {
+        paddingHorizontal: 5,
+        borderRightColor: colors.border,
+        paddingTop: insets.top + tabBarHeight,
+        borderRightWidth: StyleSheet.hairlineWidth,
+      },
+      content: {
+        flex: 1,
+        width: tabBarHeight,
+        maxWidth:
+          Platform.OS === ('ios' || 'android')
+            ? DEFAULT_TABBAR_HEIGHT
+            : DEFAULT_TABBAR_HEIGHT * 1.3,
```

Comment:
Why are both `width` and `maxWidth` needed? Also, where does `1.3` come from?

Reply:
the tabBar uses the calculated tabBarHeight as its width.
the reason for adding a minor restriction to its width at the max `DEFAULT_TABBAR_HEIGHT` which is 49 is to prevent the tabBar from extending further than that.
this is because if the width grows more than that, it takes a very large part of the screen which makes the content screen small and not so presentable.
Also, the calculation for the 1.3 for web etc was to increase the maxWidth for web since there screens are way bigger but I can make this a constant for better understanding rather than multiplying a random number

Reply:
> Also, the calculation for the 1.3 for web etc was to increase the maxWidth for web since there screens are way bigger

We shouldn't have different values for Web & native. The screen size can be big on native apps too (e.g. on a tablet). The use case for this feature is that user will use tabs on the side mainly on bigger screens and use bottom tabs on smaller screens.

Reply:
got you